connectivity/manifests/client-egress-to-echo-expression-deny.yaml

```diff
@@ -0,0 +1,32 @@
+apiVersion: cilium.io/v2
+kind: CiliumNetworkPolicy
+metadata:
+  namespace: cilium-test
+  name: client-egress-to-echo-expression
+spec:
+  endpointSelector:
+    matchExpressions:
+      - {key: 'kind', operator: Exists}
+  egressDeny:
+  - toPorts:
+    - ports:
+      - port: "8080"
+        protocol: TCP
+    toEndpoints:
+    - matchLabels:
+        io.kubernetes.pod.namespace: cilium-test
+        kind: echo
+  - toPorts:
+    - ports:
+      - port: "53"
+        protocol: ANY
+    toEndpoints:
+    - matchLabels:
+        io.kubernetes.pod.namespace: kube-system
+        k8s-app: kube-dns
+    - matchLabels:
+        io.kubernetes.pod.namespace: kube-system
+        k8s-app: coredns
+    - matchLabels:
+        io.kubernetes.pod.namespace: kube-system
+        k8s-app: node-local-dns
```

Comment:
This is a good opportunity to test `matchExpressions`, something like (untested):
```yaml
    - matchLabels:
        io.kubernetes.pod.namespace: kube-system
      matchExpressions:
      - key: "k8s-app"
        op: "equals"
        values:
         - "kube-dns"
         - "coredns"
         - "node-local-dns"
```

Reply:
good point, let me update as per the above.

I have another case testing with matchExpressions, but with exists function as well.

Reply:
thanks, op should be `In`, then it's working as expected.